### ISSUE TITLE
Bug? – "Sizes of tensors must match except in dimension 1. Expected size 28 but got size 27 for tensor number 1"

### ISSUE THREAD

**Issue by marckohlbrugge** [cou929/stability-sdk-ruby#10]
```
STABILITY_SDK_API_KEY=foobar123 stability-client --init_image=/path/to/image/dress.jpg   "beautiful painting"
```

Returns  the following error:

<code>
/Users/marc/.asdf/installs/ruby/3.0.3/lib/ruby/gems/3.0.0/gems/grpc-1.49.1/src/ruby/lib/grpc/generic/active_call.rb:29:in `check_status': 2:Exception iterating responses: Sizes of tensors must match except in dimension 1. Expected size 20 but got size 19 for tensor number 1 in the list.. debug_error_string:{UNKNOWN:Error received from peer ipv4:104.18.21.212:443 {grpc_message:"Exception iterating responses: Sizes of tensors must match except in dimension 1. Expected size 20 but got size 19 for tensor number 1 in the list.", grpc_status:2, created_time:"2022-10-02T13:21:47.397288+07:00"}} (GRPC::Unknown)
</code>

I'm not sure whether it's a bug with this gem or it's something on the serverside at DreamStudio's API. However, it happens quite often when using `--init_image`. Some images work. Other images don't. Retrying an image that does not work continues to fail.

It seems like some of the numbers in the error message change depending on the image.

This is in the latest gem version (0.2.8).

Here's an image that fails:
[dress.jpg.zip](https://github.com/cou929/stability-sdk-ruby/files/9691889/dress.jpg.zip)

**Comment by marckohlbrugge** [comment]
Did some more googling, and I the issue. Apparently both the width and the height of the image needs to be a multiple of 64 pixels: https://github.com/CompVis/stable-diffusion/issues/60

It's unfortunate the API doesn't give more descriptive errors. I wonder if the gem can do a quick check of the input file before uploading it?

**Comment by cou929** [comment]
I've stumbled similar error in text2img. I've also googled and [found some tips](https://huggingface.co/blog/stable_diffusion#:~:text=When%20choosing%20image,the%20other%20one.):

- > Make sure height and width are both multiples of 8.
- > Going below 512 might result in lower quality images.

By the way, 

>  I wonder if the gem can do a quick check of the input file before uploading it? (and perhaps even cropping it for the user)

I agree. I think it's reasonable.

**Comment by cou929** [comment]
I've added validation of image size and published it in 0.2.10 .

**Comment by cou929** [comment]
I will close once.
Would support automatic cropping if needed.
